File: 53 Maximum subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        max_sub = nums[0]
        current_sum = 0
        for num in nums:
            if current_sum < 0:
                current_sum = 0
            current_sum += num
            max_sub = max(max_sub, current_sum)

        return max_sub

File: 53. Maximum Subarray.cs
public class Solution {
    public int MaxSubArray(int[] nums) {
        int maxSoFar = nums[0];
        int currentMax = nums[0];

        for (int i = 1; i < nums.Length; i++) {
            // Either start a new subarray at current index, or extend the current subarray
            currentMax = Math.Max(nums[i], currentMax + nums[i]);
            maxSoFar = Math.Max(maxSoFar, currentMax);
        }

        return maxSoFar;
    }
}
